tests/python/unittest/test_numpy_op.py

```diff
@@ -597,6 +597,98 @@ def _test_np_exception(func, shape, dim):
+
+    tensor_shapes = [
+        ((3, 5), (3, 5), None),  # (a_shape, w_shape, axes)
+        ((4, 5, 6), (4, 5, 6), (0, 2)),
+        ((3,), (3,), 0),
+        ((2, 3), (3,), 1),
+        ((2, 3, 4), (2,), 0),
+        ((2, 3, 4), (3,), 1),
+        ((2, 3, 4), (4,), -1),
+        ((2, 3, 4, 5), (5,), 3)
+    ]
+
+    for hybridize in [True, False]:
+        for returned in [True, False]:
+            for a_shape, w_shape, axes in tensor_shapes:
+                for dtype in ['float32', 'float64']:
+                    for is_weighted in [True, False]:
+                        test_average = TestAverage(axes, returned)
+                        if hybridize:
+                            test_average.hybridize()
+                        a = np.random.uniform(-1.0, 1.0, size=a_shape, dtype=dtype)
+                        a.attach_grad()
+                        w = None
+                        np_w = None
+                        if is_weighted:
+                            w = np.random.uniform(-1.0, 1.0, size=w_shape, dtype=dtype)
+                            w.attach_grad()
+                            np_w = w.asnumpy()
+                        np_out = _np.average(a.asnumpy(), axis=axes, weights=np_w, returned=returned)
+                        with mx.autograd.record():
+                            mx_out = test_average(a, w)
+                        rtol = 1e-3
+                        atol = 1e-4
+                        if returned:
+                            np_out, np_sum_of_weights = np_out
+                            mx_out, mx_sum_of_weights = mx_out
+                            assert_almost_equal(mx_sum_of_weights.asnumpy(), np_sum_of_weights, rtol=rtol, atol=atol)
+                        assert mx_out.shape == np_out.shape
+                        assert_almost_equal(mx_out.asnumpy(), np_out.astype(dtype), rtol=rtol, atol=atol)
+                        mx_out.backward()
+                        # Code to get reference backward value
+                        a_grad, w_grad = avg_backward(a.asnumpy(), np_w, np_out, axes)
+                        assert_almost_equal(a.grad.asnumpy(), a_grad, rtol=rtol, atol=atol)
+                        if is_weighted:
+                            assert_almost_equal(w.grad.asnumpy(), w_grad, rtol=rtol*10, atol=atol*10)
+                        
+                        # Test imperative once again
+                        np_out = _np.average(a.asnumpy(), weights=np_w, axis=axes, returned=returned)
+                        mx_out = np.average(a, weights=w, axis=axes, returned=returned)
+                        if returned:
+                            np_out, np_sum_of_weights = np_out
+                            mx_out, mx_sum_of_weights = mx_out
+                            assert_almost_equal(mx_sum_of_weights.asnumpy(), np_sum_of_weights, rtol=rtol, atol=atol)
+                        assert_almost_equal(mx_out.asnumpy(), np_out.astype(dtype), rtol=rtol, atol=atol)
```

Comment:
Same here.
One more blank line below this line.
2-line gaps between Python functions recommended.